fog/fog_types/src/view.rs

```diff
@@ -190,69 +193,178 @@ pub struct TxOutRecord {
-            },
             target_key: CompressedRistrettoPublic::try_from(&self.tx_out_target_key_data[..])?,
             public_key: CompressedRistrettoPublic::try_from(&self.tx_out_public_key_data[..])?,
+            amount_masked_value: self.tx_out_amount_masked_value,
+            amount_commitment_data_crc32: crc32::checksum_ieee(&self.tx_out_amount_commitment_data),
         })
     }
 }
 
-// FogTxOut is a redacted version of the TxOut, removing the fog hint.
-// The hint is only used during ingest, so we don't need to persist it.
-#[derive(Clone, Eq, Hash, PartialEq, Message)]
+// FogTxOut is a redacted version of the TxOut, removing the fog hint, and with
+// reduced data about Amount. The hint is only used during ingest, so we don't
+// need to persist it.
+#[derive(Clone, Eq, Hash, PartialEq, Debug, Default)]
 pub struct FogTxOut {
-    /// The amount being sent.
-    #[prost(message, required, tag = "1")]
-    pub amount: Amount,
-
     /// The one-time public address of this output.
-    #[prost(message, required, tag = "2")]
     pub target_key: CompressedRistrettoPublic,
 
     /// The per output tx public key
-    #[prost(message, required, tag = "3")]
     pub public_key: CompressedRistrettoPublic,
+
+    /// The tx out masked amount
+    pub amount_masked_value: u64,
+
+    /// The crc32 of the tx out amount commitment bytes
+    pub amount_commitment_data_crc32: u32,
 }
 
+// Convert a TxOut to a FogTxOut in the efficient way (omitting compressed
+// commitment)
 impl core::convert::From<&TxOut> for FogTxOut {
     #[inline]
     fn from(src: &TxOut) -> Self {
         Self {
-            amount: src.amount.clone(),
             target_key: src.target_key,
             public_key: src.public_key,
+            amount_masked_value: src.amount.masked_value,
+            amount_commitment_data_crc32: crc32::checksum_ieee(
+                src.amount.commitment.point.as_bytes(),
+            ),
         }
     }
 }
 
-impl core::convert::From<&FogTxOut> for TxOut {
-    #[inline]
-    fn from(src: &FogTxOut) -> Self {
-        Self {
-            amount: src.amount.clone(),
-            target_key: src.target_key,
-            public_key: src.public_key,
-            e_fog_hint: EncryptedFogHint::from(&[0u8; ENCRYPTED_FOG_HINT_LEN]),
+impl FogTxOut {
+    /// Try to recover a TxOut from a FogTxOut and the user's private view key.
+    ///
+    /// * The amount commitment data is reconstructed, then we check if the
+    ///   reconstructed data matches the crc32 provided.
+    /// * The encrypted fog hint data is zeroed since it is not reconstructible
+    ///   and isn't needed by the client.
+    ///
+    /// Arguments:
+    /// * view_key: the private view key of the recipient of this TxOut
+    ///
+    /// Returns:
+    /// * TxOut,
+    /// Or
+    /// * An error if recovery failed
+    pub fn try_recover_tx_out(&self, view_key: &RistrettoPrivate) -> Result<TxOut, FogTxOutError> {
```

Comment:
Suggest renaming method to "try_recover_tx_out_fogtxtout" which would provide a more descriptive name to understand the functionality

Reply:
that name seems repetitive to me -- this is a member function on `FogTxOut` object already, so the function can already be invoked like `FogTxOut::try_recover_tx_out`